contracts/gas/GasEstimate.sol

```diff
@@ -0,0 +1,136 @@
+     * @param l1GasPrice The gas price on the source chain
+     * @return l1DataFee The L1 to L2 data fee
+     */
+    function computeExtraFee(
+        ExtraFeeType feeType,
+        bytes calldata payload,
+        uint256 l1GasPrice
+    ) internal pure returns (uint256) {
+        if (feeType == ExtraFeeType.OptimismEcotone) {
+            return optimismEcotoneL1Fee(payload, l1GasPrice);
+        }
+
+        revert UnsupportedExtraFeeType(feeType);
+    }
+
+    /**
+     * @notice Computes the L1 to L2 fee for a contract call on the Optimism chain.
+     * @param payload The payload of the contract call
+     * @param l1GasPrice The base fee for L1 to L2
+     * @return l1DataFee The L1 to L2 data fee
+     */
+    function optimismEcotoneL1Fee(bytes calldata payload, uint256 l1GasPrice)
+        internal
+        pure
+        returns (uint256 l1DataFee)
+    {
+        /* Optimism Ecotone gas model https://docs.optimism.io/stack/transactions/fees#ecotone
+             tx_compressed_size = ((count_zero_bytes(tx_data) * 4 + count_non_zero_bytes(tx_data) * 16)) / 16
+             weighted_gas_price = 16 * base_fee_scalar*base_fee + blob_base_fee_scalar * blob_base_fee
+             l1_data_fee = tx_compressed_size * weighted_gas_price
+
+           Reference implementation:
+             https://github.com/ethereum-optimism/optimism/blob/876e16ad04968f0bb641eb76f98eb77e7e1a3e16/packages/contracts-bedrock/src/L2/GasPriceOracle.sol#L138
+        */
+
+        // The new base_fee_scalar is currently set to 0.001368
+        // We are setting it to un upper bound of 0.0015 to account for possible fluctuations
+        uint256 scalarPrecision = 10**6;
+        uint256 baseFeeScalar = 1500; // 1500 : 1e6 = 0.0015
+
+        // The blob_base_fee_scalar is currently set to 0.810949. Setting it to 0.9 as an upper bound
+        // https://eips.ethereum.org/EIPS/eip-4844
+        uint256 blobBaseFeeScalar = 900000; // 0.9 multiplied by scalarPrecision
+        uint256 blobBaseFee = 1;
+
+        // Calculating transaction size in bytes that will later be divided by 16 to compress the size
+        // 68 bytes for the TX RLP encoding overhead
+        uint256 txSize = 68 * 16;
+        // GMP executeWithToken call parameters
+        // 4 bytes for method selector, 32 bytes for the commandId, 96 bytes for the sourceChain, 128 bytes for the sourceAddress, 96 bytes for token symbol, 32 bytes for amount
+        // Expecting most of the calldata bytes to be zeroes. So multiplying by 8 as a weighted average of 4 and 16
+        txSize += (4 + 32 + 96 + 128 + 96 + 32) * 8;
+
+        for (uint256 i; i < payload.length; ++i) {
+            if (payload[i] == 0) {
+                txSize += 4; // 4 for each zero byte
+            } else {
+                txSize += 16; // 16 for each non-zero byte
+            }
+        }
+
+        uint256 weightedGasPrice = 16 * baseFeeScalar * l1GasPrice + blobBaseFeeScalar * blobBaseFee;
```

Comment:
`l1GasPrice` is in terms of the src token price, whereas `blobBaseFee` hasn't been converted, so the addition is between different units :(

Sounds like it'll be easier to add a `relativeBlobBaseFee` param on the gas info at this rate?